---
title: "Week1.Rmd"
output: html_document
date: '2024-03-01'
---

Use this notebook to quickly write the methods for the week:

Source of Data
The mRNAseq samples corresponding to developmental stages P0, P4, P7, and Alzheimer's disease (AD) in vivo myocytes were systematically retrieved from the GEO database under accession number GSE64403. 

Quality Control
The initial quality control was conducted using FastQC (v0.12.0). 

Read Trimming and Cleaning
Sequencing reads underwent trimming to remove adapter sequences and low-quality bases. This preprocessing was executed using Trimmomatic (v0.39), employing a set of carefully selected parameters (ILLUMINACLIP:{adapters}:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15).

Reference Genome Indexing, Alignment and Gene Counting
Bowtie2 (v2.4.2) index was constructed for the gencode mouse primary assembly genome (GRCm39, release m34). Subsequently, reads were aligned to this reference using the STAR aligner (v2.7.11b) with default parameters. Aligned reads were quantified to gene counts using VERSE (v0.1.5) with default parameters and the gencode m34 primary assembly GTF file. 

Differential Expression Analysis
Gene counts were aggregated and filtered to exclude genes not expressed across all samples. Normalization and differential expression analysis between specific timepoints (AD vs. P0) were performed using DESeq2 (v1.42.1) with default parameters. 


If asked, you may also include any plots or figures in this notebook as well:

