---
title: "Week4.Rmd"
output: html_document
date: '2024-03-01'
---

Use this notebook to quickly write the methods for the week:






If asked, you may also include any plots or figures in this notebook as well:

Integration with RNAseq results

Download the DESeq2 results
(GSE75070_MCF7_shRUNX1_shNS_RNAseq_log2_foldchange.txt.gz) and upload this
file to the SCC.

In a jupyter notebook, apply the same filters and cutoffs as specified in
the methods of the original paper. How many DE genes do you find? Do they
match the numbers reported in the paper?
  
1153 total DE genes with 687 up and 466 downregulated genes were found and exactly match the paper 

setwd ('/projectnb/bf528/students/jawa/project-2-bkapalli/results')

data <- read.table("GSE75070_MCF7_shRUNX1_shNS_RNAseq_log2_foldchange.txt", header = TRUE)

# Filter the DE genes based on criteria
filtered_de_genes <- data[data$padj <= 0.01 & abs(data$log2FoldChange) > 1, ]
nrow(filtered_de_genes)

significant_de_genes <- na.omit(filtered_de_genes)
nrow(significant_de_genes)

# Filter upregulated genes (log2 fold change > 1)
upregulated_genes <- significant_de_genes[significant_de_genes$log2FoldChange > 1, ]
nrow(upregulated_genes)

# Filter downregulated genes (log2 fold change < -1)
downregulated_genes <- significant_de_genes[significant_de_genes$log2FoldChange < -1, ]
nrow(downregulated_genes)


Using the list of DE genes downloaded in step 1 and the annotated peak
file you generated previously, recreate figure 2f (Bar graph showing RUNX1 peak binding +/− 5kb of transcriptional start site (TSS), or +/− 20kb of the gene bodies of up- and down-regulated genes.) and supplementary figure
S2D (Bar graph showing the percentage of RUNX1 peaks (numbers on the bars) +/- 100kb of differentially expressed genes.) (attached) and produce stacked barcharts showing the proportions of DE genes with a
Runx1 peak found within +/- 5kb, +/- 20kb, and +/- 100kb of the TSS

# Code used for creating plots

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Function to calculate the proportions and absolute numbers of DE genes with a Runx1 peak within a specified distance to the TSS
def runx1_peak_proportions(annotated_peaks, de_genes, distance):
    peaks_near_tss = annotated_peaks[annotated_peaks['Distance to TSS'].abs() <= distance]
    de_genes_with_peaks = pd.merge(de_genes, peaks_near_tss, left_on='genename', right_on='Gene Name', how='left')
    bound_genes = de_genes_with_peaks['PeakID'].notnull().sum()
    not_bound_genes = len(de_genes) - bound_genes
    proportion_bound = (bound_genes / len(de_genes)) * 100
    proportion_not_bound = (not_bound_genes / len(de_genes)) * 100
    return bound_genes, not_bound_genes, proportion_bound, proportion_not_bound

# Calculate the values for each set of genes and distances
values_5kb_up = runx1_peak_proportions(annotated_peaks_df, upregulated_genes_df, 5000)
values_5kb_down = runx1_peak_proportions(annotated_peaks_df, downregulated_genes_df, 5000)
values_20kb_up = runx1_peak_proportions(Annotated_peaks_df, upregulated_genes_df, 20000)
values_20kb_down = runx1_peak_proportions(annotated_peaks_df, downregulated_genes_df, 20000)
values_100kb_up = runx1_peak_proportions(annotated_peaks_df, upregulated_genes_df, 100000)
values_100kb_down = runx1_peak_proportions(annotated_peaks_df, downregulated_genes_df, 100000)

# Set up the matplotlib figure for three subplots
fig, axes = plt.subplots(1, 3, figsize=(21, 6))

# Define the indices for the bars
indices = np.arange(2)
bar_width = 0.5

# Function to add bars to the plot for a specific distance
def add_bars(ax, values_up, values_down, labels, distance):
    # Plot bars for 'RUNX1 bound'
    ax.bar(indices, [values_up[0], values_down[0]], bar_width, label='RUNX1 bound', color='red')
    # Plot bars for 'Not bound' on top of 'RUNX1 bound'
    ax.bar(indices, [values_up[1], values_down[1]], bar_width, label='Not bound', color='grey')

    # Add count labels for 'RUNX1 bound' and 'Not bound'
    ax.text(indices[0], values_up[2] / 2, f'{values_up[0]}', ha='center', va='center', color='white')
    ax.text(indices[1], values_down[2] / 2, f'{values_down[0]}', ha='center', va='center', color='white')
    ax.text(indices[0], values_up[2] + (values_up[3] - values_up[2]) / 2, f'{values_up[1]}', ha='center', va='center', color='black')
    ax.text(indices[1], values_down[2] + (values_down[3] - values_down[2]) / 2, f'{values_down[1]}', ha='center', va='center', color='black')

    # Set labels and title
    ax.set_ylabel('Number of Genes')
    ax.set_xticks(indices)
    ax.set_xticklabels(labels)
    ax.set_title(f'RUNX1 peak binding {distance} of TSS')
    ax.legend()

# Adding bars to each subplot for the three distances
add_bars(axes[0], values_5kb_up, values_5kb_down, ['Up-regulated', 'Down-regulated'], '±5kb')
add_bars(axes[1], values_20kb_up, values_20kb_down, ['Up-regulated', 'Down-regulated'], '±20kb')
add_bars(axes[2], values_100kb_up, values_100kb_down, ['Up-regulated', 'Down-regulated'], '±100kb






![Fig2D,2SD](/projectnb/bf528/students/bkapalli/project-2-bkapalli/Fig2D,2SD.jpeg)






Visualizing peaks using a genome browser

Download IGV (https://software.broadinstitute.org/software/igv/) locally on your computer or use the newly added web-only interface (https://igv.org/app/). You will need to provide the following files:

1. bigWig files for the runx1_rep1, runx1_rep2, inp_rep1, inp_rep2 samples

2. Your BED file of reproducible peaks

3. The primary assembly GTF file containing the annotations for our 
  reference genome.
You will have to download these files locally, but all of them are small enough that this should cause no issues.

1. Navigate to the two genes mentioned specifically in the paper, 
  MALAT1 and NEAT1. 
    - Do you see the same general results as in figures 2d and 2e?

Indeed, the patterns in my findings align with those observed in the figures.

    - Do you agree with the conclusions made by the authors in these figures?

    Absolutely, we observe a pronounced binding of RUNX1 at the promoter area of MALAT1, with subsequent areas showing negligible peaks, indicating suppression. Figure 2E mirrors this observation for the NEAT1 lncRNA gene locus, where prominent peak regions at the promoter coincide with diminished expression of NEAT1 past this region. These findings corroborate the authors' assertion that "RUNX1's binding at the promoter regions of these crucial lncRNAs, which play significant roles in nuclear structuring and the progression of cancer, implies a direct transcriptional influence of RUNX1 on their expression." The data presented in the figures effectively demonstrate RUNX1's interaction with the promoters of MALAT1 and NEAT1, leading to the reduced expression of these lncRNA genes upon RUNX1 depletion, which supports the notion of RUNX1's direct transcriptional regulation over their expression.

Although we have discouraged you from generating figures using screenshots, we are going to produce the same figure as found in 2d and 2e.

1. Please take screenshots of the MALAT1 and NEAT1 promoter regions
  displaying the signal coverage found in your bigWig files and the peak
  locations from your BED file of reproducible peaks.

![MALAT1](/projectnb/bf528/students/bkapalli/project-2-bkapalli/MALAT1.jpeg)

![NEAT1](/projectnb/bf528/students/bkapalli/project-2-bkapalli/NEAT1.jpeg)


